vowpalwabbit/reductions/cb/cb_to_cb_adf.cc

```diff
@@ -198,26 +194,25 @@ VW::LEARNER::base_learner* cb_to_cb_adf_setup(VW::setup_base_i& stack_builder)
 
   // see csoaa.cc ~ line 894 / setup for csldf_setup
   all.example_parser->emptylines_separate_examples = false;
-  VW::prediction_type_t pred_type;
 
   if (data->explore_mode)
   {
     data->adf_learner = as_multiline(base->get_learner_by_name_prefix("cb_explore_adf_"));
-    pred_type = VW::prediction_type_t::action_probs;
   }
   else
   {
     data->adf_learner = as_multiline(base->get_learner_by_name_prefix("cb_adf"));
-    pred_type = VW::prediction_type_t::multiclass;
   }
 
   auto* l = make_reduction_learner(
       std::move(data), base, predict_or_learn<true>, predict_or_learn<false>, all.get_setupfn_name(cb_to_cb_adf_setup))
-                .set_output_prediction_type(pred_type)
                 .set_input_label_type(VW::label_type_t::cb)
+                .set_output_label_type(VW::label_type_t::cb)
+                .set_input_prediction_type(base->get_output_prediction_type())   // action_scores or action_probs
```

Comment:
We know which one it will be based on data->explore_mode. So we should set the specific type